### PR TITLE
Revert "replication: disable replication if its primary state"

### PR DESCRIPTION
**Pull request by Madhu-1**
This reverts #94. More discussions on https://github.com/csi-addons/volume-replication-operator/issues/100

### CODE REVIEW

**Review by ShyamsundarR — APPROVED** [review]

**Comment by ShyamsundarR** [comment]
Would like to see the ceph-csi changes to handle disable as secondary, so that the workflow is not disturbed, but that may come later I guess.

File: controllers/volumereplication_controller.go [hunk 1]
```diff
@@ -167,15 +167,12 @@ func (r *VolumeReplicationReconciler) Reconcile(ctx context.Context, req ctrl.Re
...
 		if contains(instance.GetFinalizers(), volumeReplicationFinalizer) {
-			// Disable replication only if it is in primary state and PV object
-			// is present.
-			if instance.Spec.ReplicationState == replicationv1alpha1.Primary && pvErr == nil {
-				err := r.disableVolumeReplication(logger, volumeHandle, parameters, secret)
-				if err != nil {
-					logger.Error(err, "failed to disable replication")
-					return ctrl.Result{}, err
-				}
+			err := r.disableVolumeReplication(logger, volumeHandle, parameters, secret)
```
Comment (ShyamsundarR) [inline controllers/volumereplication_controller.go:170]: We need to call disable without checking state of the VR in the case of Delete as well, which currently calls disable only if it is Primary, right?

Comment (Madhu-1) [reply controllers/volumereplication_controller.go:170]: Yes correct I will fix it as part of https://github.com/csi-addons/volume-replication-operator/pull/104

Comment (Madhu-1) [reply controllers/volumereplication_controller.go:170]: sorry, discard my last comment. This is reverting back the change and there is no check on the VR state to call the disableReplication. Am I missing something?

**Comment by Madhu-1** [comment]
> Would like to see the ceph-csi changes to handle disable as secondary, so that the workflow is not disturbed, but that may come later I guess.

Yes working on the cephcsi changes will send PR soon for that one.

File: controllers/volumereplication_controller.go (hunk 1 above)
Comment (ShyamsundarR) [reply controllers/volumereplication_controller.go:170]: We should call disable unconditionally [here](https://github.com/csi-addons/volume-replication-operator/blob/00f42351bccb8020ef90d36e65d4320959154285/controllers/volumereplication_controller.go#L172-L173) as well, right? As the storage provider needs to handle if it is `Primary` and do the appropriate action.

Comment (Madhu-1) [reply controllers/volumereplication_controller.go:170]: isn't it doing at the same place?

**Review by iamniting — APPROVED** [review]

File: controllers/volumereplication_controller.go (hunk 1 above)
Comment (ShyamsundarR) [reply controllers/volumereplication_controller.go:170]: yes, ack! sorry wires crossed between PR reviews.

**Pull request merged**